### ISSUE TITLE
Transactions table is not displaying

### ISSUE THREAD

**Issue by SrujanChary-2003** [adrianhajdin/banking#83]
![image](https://github.com/user-attachments/assets/9498d5e9-8765-4084-9a76-472adff94956)

An error occurred while getting the account: TypeError: transactions is not iterable
    at $$ACTION_1 (webpack-internal:///(rsc)/./lib/actions/bank.actions.ts:119:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async TransactionHistory (webpack-internal:///(rsc)/./app/(root)/transaction-history/page.tsx:46:21)
 GET /transaction-history?_rsc=1suv6 200 in 5279ms
 ⨯ components\TransactionsTable.tsx (43:23) @ map
 ⨯ TypeError: Cannot read properties of undefined (reading 'map')
    at TransactionsTable (./components/TransactionsTable.tsx:108:40)
    at stringify (<anonymous>)
digest: "3782927363"
  41 |       </TableHeader>
  42 |       <TableBody>
> 43 |         {transactions.map((t: Transaction) => {
     |                       ^
  44 |           const status = getTransactionStatus(new Date(t.date))
  45 |           const amount = formatAmount(t.amount)   this is the error displayed in terminal please provide us solution sir..

**Comment by Manoj-Kande** [comment]
your transaction table is empty. cross check the getAccount(function) and getTransactions(function)  and page.tsx(root folder).

**Comment by MOMOJR321** [comment]
i am having this same problem.

did you manage to fix it

**Comment by AdugboAghogho** [comment]
i am having this same problem.

**Comment by AdugboAghogho** [comment]
@Manoj-Kande my RecentTransactions is not displaying, and the issue is from TransactionsTable and when i continued, TransactionHistory this error show, how exactly did u fix ours, i went through your codes i couldn't find a fix.... if u don't mind pls explain

**Comment by Manoj-Kande** [comment]
> i am having this same problem.
> 
> did you manage to fix it

Yes, I made some changes to the createLinkToken function. You can refer to the code in the transaction_table.txt file in the repository. After making these changes, I created a new user and tried it, and the problem was fixed. Additionally, ensure to remove the documents in Dwolla and Appwrite if the newly created user already exists.

**Comment by Manoj-Kande** [comment]
> @Manoj-Kande my RecentTransactions is not displaying, and the issue is from TransactionsTable and when i continued, TransactionHistory this error show, how exactly did u fix ours, i went through your codes i couldn't find a fix.... if u don't mind pls explain

You can refer to this solution to fix your problem
https://github.com/adrianhajdin/banking/issues/83#issuecomment-2561890431

**Comment by AdugboAghogho** [comment]
thanks so much... it worked!!!

**Comment by Ayush-1127** [comment]
Here is the Problem fixed
Do changes in the function createLinkToken which is present in the file name user.actions.ts.

export const createLinkToken = async (user: User) => {
  try {
    const tokenParams = {
      user: {
        client_user_id: user.$id
      },
      client_name: `${user.firstName} ${user.lastName}`,
      products: ['auth', 'transactions'] as Products[],
      language: 'en',
      country_codes: ['US'] as CountryCode[],
    }

    const response = await plaidClient.linkTokenCreate(tokenParams);

    return parseStringify({ linkToken: response.data.link_token })
  } catch (error) {
    console.log(error);
  }
}
